### PR TITLE
feat(aggregations): always display $search and add Atlas CTA

### PR DESCRIPTION
**Pull request by addaleax**
Always include `$search` in the list of aggregation stages,
displayed together with an Atlas logo for indicating
Atlas-only support, and add an Atlas signup CTA for cases
in which the `$search` stage fails because it is unsupported.

Co-authored-by: @gribnoysup 

Screenshot from 2021-09-06 17-29-14

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by gribnoysup** [comment]
It _just_ hit me that migh've forgot to check the ADL case 🙈

File: packages/compass-aggregations/src/components/stage-preview/stage-preview.jsx [hunk 1]
```diff
@@ -52,6 +55,13 @@ class StagePreview extends Component {
     this.props.runOutStage(this.props.index);
   }
 
+  /**
+   * Called when the Atlas Signup CTA link is clicked.
+   */
+  onCreateAtlasCluster = () => {
+    this.props.openLink('https://www.mongodb.com/cloud/atlas/lp/general/try?utm_source=compass&utm_medium=product');
```
Comment (addaleax) [inline packages/compass-aggregations/src/components/stage-preview/stage-preview.jsx:62]: (note: we need to wait for @mmarcon for the proper link here before merging this)

Comment (mmarcon) [reply packages/compass-aggregations/src/components/stage-preview/stage-preview.jsx:62]: I requested how to request it :D I'll know more once the US wakes up.

**Review by mcasimir — APPROVED** [review]
So cool!

File: packages/compass-aggregations/src/components/stage-preview/stage-preview.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (addaleax) [reply packages/compass-aggregations/src/components/stage-preview/stage-preview.jsx:62]: @mcasimir Done :)

File: packages/compass-aggregations/src/components/stage-preview/stage-preview.jsx [hunk 2]
```diff
@@ -132,12 +142,39 @@ class StagePreview extends Component {
     );
   }
 
+  /**
+   * If the stage operator is $search and it is not supported we
+   * show a Atlas signup CTA.
+   *
+   * @returns {Component} The component.
+   */
+  renderMissingStageSupportSection() {
```
[the remark this answers is not in the document]
Comment (addaleax) [reply packages/compass-aggregations/src/components/stage-preview/stage-preview.jsx]: @mcasimir Done (but with the `Section` suffix for the method name for consistency with the other panels

[the remark this answers is not in the document]
Comment (mcasimir) [reply packages/compass-aggregations/src/components/stage-preview/stage-preview.jsx]: Ah ok makes sense!

**Pull request merged**